src/validation.cpp

```diff
@@ -2600,6 +2600,14 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
              Ticks<SecondsDouble>(m_chainman.time_forks),
              Ticks<MillisecondsDouble>(m_chainman.time_forks) / m_chainman.num_blocks_total);
 
+    {
+        static bool script_check_started{fScriptChecks}; // Only log if we start off with disabled script checks
+        if (fScriptChecks && !script_check_started) {
```

Comment:
fScriptChecks isn't a one-way gate. It would be very confusing to see "Started validation" due to a block not an ancestor of the assumevalid block, and then have validation of the real chain continue without validation and no log informing the user it's being skipped again.